protected_tags.go

```diff
@@ -0,0 +1,141 @@
+package gitlab
+
+import (
+	"fmt"
+	"net/url"
+)
+
+// ProtectedTagsService handles communication with the protected tag methods of the GitLab API.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/protected_tags.html
+type ProtectedTagsService struct {
+	client *Client
+}
+
+// TagAccessDescription reperesents the access decription for a protected tag.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/protected_tags.html#list-protected-tags
+type TagAccessDescription struct {
+	AccessLevel            AccessLevelValue `json:"access_level"`
+	AccessLevelDescription string           `json:"access_level_description"`
+}
+
+// ProtectedTag represents a protected tag.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/protected_tags.html#list-protected-tags
```

Comment:
The URL should probably omit `#list-protected-tags` as this is the generic type that is used by all endpoints.